src/Serialization/Fields.jl

```diff
@@ -48,3 +48,132 @@ function load_internal(s::DeserializerState, ::Type{gfp_fmpz_elem}, dict::Dict)
     F = load_type_dispatch(s, Nemo.GaloisFmpzField, dict[:parent])
     return F(load_type_dispatch(s, fmpz, dict[:data]))
 end
+
+################################################################################
+# SimpleNumField
+function save_internal(s::SerializerState, K::SimpleNumField)
+    return Dict(
+        :def_pol => save_type_dispatch(s, defining_polynomial(K)),
+        :var => save_type_dispatch(s, String(var(K)))
+    )
+end
+
+function load_internal(s::DeserializerState, ::Type{<: SimpleNumField}, dict::Dict)
+    def_pol = load_type_dispatch(s, dict[:def_pol], check_namespace=false)
+    var = load_type_dispatch(s, dict[:var], check_namespace=false)
+    K, _ = NumberField(def_pol, var, cached=false)
+
+    return K
+end
+
+################################################################################
+# FqNmodfinitefield
+function save_internal(s::SerializerState, K::FqNmodFiniteField)
+    return Dict(
+        :def_pol => save_type_dispatch(s, defining_polynomial(K))
+    )
+end
+
+function load_internal(s::DeserializerState, ::Type{FqNmodFiniteField}, dict::Dict)
+    def_pol = load_type_dispatch(s, dict[:def_pol], check_namespace=false)
+    K, _ = FiniteField(def_pol, cached=false)
+
+    return K
+end
+
+#elements
+function save_internal(s::SerializerState, k::Union{nf_elem, fq_nmod, Hecke.NfRelElem})
+    K = parent(k)
+    polynomial = parent(defining_polynomial(K))(k)
+    K_dict = save_type_dispatch(s, K)
+
+    return Dict(
+        :parent => K_dict,
+        :polynomial => save_type_dispatch(s, polynomial)
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{nf_elem, fq_nmod, Hecke.NfRelElem}},
+                       dict::Dict)
+    K = load_type_dispatch(s, dict[:parent], check_namespace=false)
```

Comment:
Again I wonder if perhaps a combination of `dense_poly_type` and `elem_type` could be used to deduce the type of the polynomial precisely?

(Not saying this is necessary, feel free to ignore)